Vpngw generation add

AdamBrassard opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
I needed the ability to set the Generation of the VirtualNetworkGateway and wanted to contribute this to the community.
Adding the ability to set Virtual Network Gateway Generation in azure_rm_virtualnetworkgateway module. 

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request


##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
plugins/module_utils/azure_rm_common.py
plugins/modules/azure_rm_virtualnetworkgateway.py
tests/integration/targets/azure_rm_virtualnetworkgateway/tasks/main.yml

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
- Adding support for setting VPNGW generation
- Updated azure_rm_common.py NetworkManagementClient version to 2019-11-01
- Updated azure_rm_virtualnetworkgateway.py for additional input and choices of VPNGW Generation options as outlined in SDK
- Updated integration test to generate public ip name in same method as vngname for multiple runs
- Updated integration tests adding block for gen setting and clean up of VNET/SUBNET/PUBIP after test and delete of VPNGW

<!--- Paste verbatim command output below, e.g. before and after your change -->
```yaml
- name: Create virtual network gateway without bgp settings
  azure_rm_virtualnetworkgateway:
    resource_group: "{{ resource_group }}"
    name: "{{ vngname }}"
    ip_configurations:
      - name: testipconfig
        private_ip_allocation_method: Dynamic
        public_ip_address_name: "{{ pubipname }}"
    virtual_network: "{{ vnetname }}"
    tags:
      common: "xyz"
```
```yaml
- name: Create virtual network gateway Generation2 (check mode)
  check_mode: yes
  azure_rm_virtualnetworkgateway:
    resource_group: "{{ resource_group }}"
    name: "{{ vngname }}"
    sku: VpnGw2
    vpn_gateway_generation: Generation2
    ip_configurations:
      - name: testipconfig
        private_ip_allocation_method: Dynamic
        public_ip_address_name: "{{ pubipname }}" #testPublicIP
    virtual_network: "{{ vnetname }}"
    tags:
      common: "xyz"
```

AdamB-cc committed:
Adding Support for VirtualNetworkGateway Generation setting.
Updated version of NetworkManagementClient to reflect additional functionality

AdamB-cc committed:
Adding fucntion to integration tests
Cleaning trailing whitespace from py file

Fred-sun commented on CHANGELOG.md:
```diff
@@ -1,5 +1,10 @@
 # Change Log
 
+## v1.13.1 (2022-07-14)
```
Please delete 3~7 lines. It is not a major change and will not be released for a separate feature. So please cancel, we will regularly post all recent changes together. Thank you very much!

AdamBrassard replied:
Of course, thank you. This has been removed

AdamBrassard committed:
Update CHANGELOG.md

removing changelog entry as requested as it is not a major change and will not be released for a separate feature.

Fred-sun commented on galaxy.yml:
```diff
@@ -9,7 +9,7 @@ namespace: azure
 
 # The version of the collection. Must be compatible with semantic versioning
-version: 1.13.0
+version: 1.13.1
```
Now we release a version regularly, the next version should be 1.14.0. Thank you!
```suggestion
version: 1.13.0
```

AdamBrassard committed:
Update galaxy.yml to follow version standard

Co-authored-by: Fred-sun <37327967+Fred-sun@users.noreply.github.com>

xuzhang3 commented:
@AdamBrassard LGTM
